Feature/#15 post create

mh-algo opened the pull request:
## #️⃣연관된 이슈

> #15

## 📝작업 내용

> 작성된 게시물 저장 구현
> 테스트 코드 작성

### 스크린샷 (선택)

## 💬리뷰 요구사항(선택)

mh-algo committed:
feat: #15 게시판 작성 repository 추가

mh-algo committed:
feat: #15 게시판 작성 controller, service 추가

mh-algo committed:
feat: #15 검증 추가

mh-algo committed:
refactor: #15 PostCreateReq 수정

- 테스트 코드에서 객체 비교가 필요해서 @EqualsAndHashCode 추가

mh-algo committed:
test: #15 PostControllerTest 추가

Yuurim98 reviewed (approved):
고생하셨습니다!

seongminQa reviewed (approved):
고생하셨습니다!! 도와주셔서 감사합니다 ㅎㅎ

dy1214 reviewed (approved):
확인했습니다!

0515-Yoonseo-Kim reviewed (approved):
승인했습니다.

mh-algo committed:
refactor: #15 코드 리팩터링

- PostController에서 @RequestMapping 사용하도록 수정
- PostCreateReq의 @NotNull 제거(@NotBlank와 역할이 겹침)

mh-algo committed:
test: #15 통합테스트 추가

mh-algo committed:
refactor: #15 코드 리팩터링

- Mapper xml 파일 통합
- PostService의 save 메서드에서 mapper 쓰는 방식으로 통일

mh-algo committed:
refactor: #15 중복 등록된 import 제거

mh-algo committed:
test: #15 작성한 post가 db에 저장되었는지 확인하는 과정 추가
